float64 t
float64 v
float64 theta
